pkg/sqlite/graphloader.go

```diff
@@ -0,0 +1,134 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/operator-framework/operator-registry/pkg/registry"
+)
+
+// GraphLoader generates a graph
+// GraphLoader supports multiple different loading schemes
+// GraphLoader from SQL, GraphLoader from old format (filesystem), GraphLoader from SQL + input bundles
+type GraphLoader interface {
+	Generate() (*registry.Package, error)
+}
+
+type SQLGraphLoader struct {
+	Querier     *SQLQuerier
+	PackageName string
+}
+
+func NewSQLGraphLoader(dbFilename, name string) (*SQLGraphLoader, error) {
+	querier, err := NewSQLLiteQuerier(dbFilename)
+	if err != nil {
+		return nil, err
+	}
+
+	return &SQLGraphLoader{
+		Querier:     querier,
+		PackageName: name,
+	}, nil
+}
+
+func NewSQLGraphLoaderFromDB(db *sql.DB, name string) (*SQLGraphLoader, error) {
+	return &SQLGraphLoader{
+		Querier:     NewSQLLiteQuerierFromDb(db),
+		PackageName: name,
+	}, nil
+}
+
+func (g *SQLGraphLoader) Generate() (*registry.Package, error) {
+	ctx := context.TODO()
+	defaultChannel, err := g.Querier.GetDefaultPackage(ctx, g.PackageName)
+	if err != nil {
+		return nil, err
+	}
+
+	channelEntries, err := g.Querier.GetChannelEntriesFromPackage(ctx, g.PackageName)
+	if err != nil {
+		return nil, err
+	}
+
+	channels, err := g.GraphFromEntries(channelEntries)
+	if err != nil {
+		return nil, err
+	}
+
+	return &registry.Package{
+		Name:           g.PackageName,
+		DefaultChannel: defaultChannel,
+		Channels:       channels,
+	}, nil
+}
+
+// GraphFromEntries builds the graph from a set of channel entries
+func (g *SQLGraphLoader) GraphFromEntries(channelEntries []registry.ChannelEntryNode) (map[string]registry.Channel, error) {
```

Comment:
I'm wondering if we should have`GraphFromEntries` be on the `GraphLoader` interface as well - we want all the different loaders to be able to provide this functionality as well right

Reply:
Or maybe we can have a separate `Grapher` interface 😄

Reply:
I actually think we explicitly don't want other graphloaders to have this method. This is really specific to the sqlite layer, other loaders wouldn't be aware of this aggregation at all.